.github/workflows/main.yml

```diff
@@ -71,25 +71,14 @@ jobs:
         brew update && brew install coreutils swig ninja
       if: matrix.tar == 'osx'
       shell: bash
-    - name: Install Linux tools for LLDB
-      run: |
-        sudo apt-get install build-essential swig python3-dev libedit-dev libncurses5-dev libxml2-dev
-      if: matrix.os == 'ubuntu-22.04'
     - name: Free Disk Space (Ubuntu)
       uses: jlumbroso/free-disk-space@main
       if: matrix.os == 'ubuntu-22.04'
       with:
         tool-cache: false
-    - name: Install Ninja
-      run: |
-        curl -L -O "https://github.com/ninja-build/ninja/releases/download/v1.12.1/${{ matrix.ninja_file }}" && \
-        ${{ matrix.ninja_sudo }} unzip -o ${{ matrix.ninja_file }} -d ${{ matrix.ninja_dir }} && rm ${{ matrix.ninja_file }}
-      if: matrix.tar != 'osx'
-      shell: bash
-    - name: Install Rust
-      uses: dtolnay/rust-toolchain@master
+    - uses: cachix/install-nix-action@v31
       with:
-        toolchain: 1.84.1
+        nix_path: nixpkgs=https://github.com/NixOS/nixpkgs/archive/3250c94d987c5ebcf24a5bfcd3c7de5ec43e3e84.tar.gz
     - name: Build
       run: ./build.sh ${{ matrix.out_dir }}
```

Comment:
I see you are not passing the `--nix` flag here. Is you idea to build the release binaries using nix?